internal/deepprint/deepprint.go

```diff
@@ -45,6 +49,69 @@ func print(w *bufio.Writer, v reflect.Value, visited map[uintptr]bool) {
 	if !v.IsValid() {
 		return
 	}
+
+	// Special case some common types.
+	if v.CanInterface() {
+		switch v.Type() {
+		case reflect.TypeOf(netaddr.IP{}):
+			var b []byte
+			var err error
+			if v.CanAddr() {
+				x := v.Addr().Interface().(*netaddr.IP)
+				b, err = x.MarshalText()
+			} else {
+				x := v.Interface().(netaddr.IP)
+				b, err = x.MarshalText()
+			}
+			if err == nil {
+				w.Write(b)
+				return
+			}
+		case reflect.TypeOf(netaddr.IPPrefix{}):
+			var b []byte
+			var err error
+			if v.CanAddr() {
+				x := v.Addr().Interface().(*netaddr.IPPrefix)
+				b, err = x.MarshalText()
+			} else {
+				x := v.Interface().(netaddr.IPPrefix)
+				b, err = x.MarshalText()
+			}
+			if err == nil {
+				w.Write(b)
+				return
+			}
+		case reflect.TypeOf(wgkey.Key{}):
```

Comment:
instead of special-casing these three types, look for any array of byte instead like you did for reflect.DeepEqual recently?

Reply:
I did this a few ways and couldn't get the performance to be comparable. I'm going to merge this PR for now, since it is a clear improvement, and then keep tinkering.